packages/README.md

```diff
@@ -0,0 +1,73 @@
+# Package Metadata Description
+
+## 1. Abstract
+Working with many differing artifacts coming out of many differing package ecosystems requires us to find common ground between all
+the ecosystems. While *requires* may have been a little strong wording to start with, let's overview the benefits of having
+a common metadata description for all ecosystems:
+
+ * Metadata storage - Database
+   * All data are stored in generic rows
+   * There' onlys one main table
+   * Database can operate with the metadata directly
+ * Common codebase
+   * One API, one Interchange format
+   * One binding for all client languages
+ * Upstreams can change their metadata format on a whim
+
+While some points are important during development, especially the last bullet is very important for long term maintenance, since it 
+allows us to `contain` the volatile upstream specification within single minimal component for each ecosystem. Let's see an example:
+
+```
+         +-------------------+
+IN       |                   |  OUT
+         |  npm metadata     |
++------->+  manifest tool    +------------>
+         |                   |
+         +-------------------+
+```
+
+`IN`put is a `npm` package and `OUT`put is a document in common metadata format. Whenever the `npm` upstream changes its package specification
```

Comment:
Yes, optimally, the input archive is digested using `sha` or whatever and that is used as an unique identifier within our infrastructure.